### ISSUE TITLE
A Meta-learning Approach to Fair Ranking

### ISSUE THREAD

**Issue by kmaahs001** [kmaahs001/CapstoneResources#8]
### Title

A Meta-learning Approach to Fair Ranking

### URL

https://www.cse.scu.edu/~yfang/Meta_Learning_Fairness_SIGIR_2022.pdf

### Summary 

Meta-learning based Fair Ranking (MFR) model that alleviates data bias for protected groups using an automatically weighted loss. Specifically, they use a meta-learning framework to **explicity train a meta-learner from an unbiased sampled dataset (meta-dataset)** and simultaneously train a **listwise LTR** model on the whole (biased) dataset governed by "fair" loss weights. 

The meta-learner serves as a weighting function to make the ranking loss attend more on the minority group.

### Key Points 

- Although the in-processing models have achieved good performance on the fairness metric, there is still the limitation as the model is learned from the biased dataset. Thus, the meta-learning could benefit the aforementioned problem by training a meta-learner on a meta-dataset.
- Meta-dataset is collected uniformly without any bias, which would train a fair meta-learner so that the ranking model could learn from it.

Overview:

- Meta-learning framework to formulate the fairness-aware ranking task as a bilevel optimization problem, where the upper-level is the meta-trainer and the lower-level is the ranking model.
- Train a meta-learner on the meta-dataset which could help the ranking model to learn fairly on the biased dataset.
- The meta-dataset is a small unbiased dataset, which is collected by uniformly sampling from the training dataset under all queries for both the protected group and the unprotected group. 
- In detail, at each training iteration, we use the ranking model and the ranking loss function to compute the loss values for each data sample from the training dataset. 
- Then we train a multi-layer neural network as the weighting function to re-weight the loss values, and the weighting function is optimized by the weighted loss values on the meta-dataset. 
- Since the weighting function which is the meta-learner is subject to the ranking models, our goal is to optimize the loss’ weights (given by the meta-learner) to achieve fairness on the training dataset.
-  Intuitively, we can see the loss’ weight as the hyperparameter which could be learned, and we train a meta-learner to tune the **hyperparameter** on the meta-dataset.
-  Such the training process could also be referred to as the bilevel optimization as the learned parameters of the ranking model depend on the parameters of the meta-learner. 

### Citation

Yuan Wang, Zhiqiang Tao, and Yi Fang. 2022. A Meta-learning Approach to Fair Ranking. In Proceedings of the 45th International ACM SIGIR Conference on Research and Development in Information Retrieval (SIGIR ’22), July 11–15, 2022, Madrid, Spain. ACM, New York, NY, USA, 6 pages. https://doi.org/10.
1145/3477495.3531892
